cmd/kubefwd/services/services.go

```diff
@@ -434,6 +436,60 @@ func getEnv(key, fallback string) string {
 	return fallback
 }
 
+type HeadlessServiceWatcher interface {
+	WatchHeadlessService(stopChannel <-chan struct{}, service *v1.Service)
+}
+
+type HeadlessServiceWaiterImpl struct {
+	Namespace   string
+	ServiceName string
+	ClientSet   kubernetes.Clientset
+	ServiceFwd  *fwdservice.ServiceFWD
+}
+
+func (waiter *HeadlessServiceWaiterImpl) WatchHeadlessService(stopChannel <-chan struct{}, service *v1.Service) {
+
+	watcher, err := waiter.ClientSet.CoreV1().Endpoints(waiter.Namespace).Watch(
+		context.TODO(), metav1.SingleObject(metav1.ObjectMeta{Name: service.ObjectMeta.Name, Namespace: service.ObjectMeta.Namespace}))
+	if err != nil {
+		log.Errorf("Cannot watch endpoints for service %s in namespace %s", service.ObjectMeta.Name, waiter.Namespace)
+		return
+	}
+
+	// Listen for stop signal from above
+	go func() {
+		<-stopChannel
+		watcher.Stop()
+	}()
+
+	// watcher until the pod is deleted, then trigger a syncpodforwards
+	for {
+		event, ok := <-watcher.ResultChan()
+		if !ok {
+			log.Errorf("ENDPOINT: Couldn't receive message")
+			return
+		}
+		log.Warnf("Event: %s", event.Type)
+		switch event.Type {
+		case watch.Added: // add the tunnel
+			log.Warnf("ADDED ENDPOINT: Service %s, Endpoint added. ", service.ObjectMeta.Name)
```

Comment:
I think this should not be a warning or error, just log the statement.

Reply:
done